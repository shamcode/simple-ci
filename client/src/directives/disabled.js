export default class {
    constructor() {
        this.node = null;
    }

    bind( node ) {
        this.node = node;
    }

    unbind() {
        this.node = null;
    }

    update( value ) {
        this.node.disabled = value;
    }
}
