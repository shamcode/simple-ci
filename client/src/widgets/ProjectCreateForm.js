import { options } from 'sham-ui';
import ProjectCreateFormTemplate from './ProjectCreateForm.sht';

export default class ProjectCreateForm extends ProjectCreateFormTemplate {
    @options get dataSaving() { return false; }
    @options onSubmit() {}

    get formNode() {
        return this.querySelector( 'form' );
    }

    _submitForm( e ) {
        e.preventDefault();
        const formData = new FormData( this.formNode );
        const data = {};
        formData.forEach( ( value, key ) => data[ key ] = value );
        this.options.onSubmit( data );
    }
}