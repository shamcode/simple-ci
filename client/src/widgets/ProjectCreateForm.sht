<div class="project-create-form">
    <form>
        <div class="group">
            <input name="name" type="text" required/>
            <span class="highlight"></span>
            <span class="bar"></span>
            <label>Name</label>
        </div>
        <div class="group">
            <input name="cwd" type="text" required/>
            <span class="highlight"></span>
            <span class="bar"></span>
            <label>Current Work Directory</label>
        </div>
        <div class="group">
            <button type="submit" :disabled={{dataSaving}} >
                <span>Create</span>
            </button>
        </div>
    </form>
</div>
