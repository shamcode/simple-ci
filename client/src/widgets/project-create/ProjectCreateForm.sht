<div class="project-create-form">
    <form :onsubmit={{this._submitForm.bind(this)}}>
        <div class="group">
            <input name="name" type="text" required value={{name}}/>
            <span class="highlight"></span>
            <span class="bar"></span>
            <label>Name</label>
        </div>
        <div class="group">
            <input name="cwd" type="text" required value={{cwd}}/>
            <span class="highlight"></span>
            <span class="bar"></span>
            <label>Current Work Directory</label>
        </div>
        <div class="group">
            <button type="submit" :disabled={{dataSaving}} >
                <span>Create</span>
            </button>
        </div>
    </form>
</div>
