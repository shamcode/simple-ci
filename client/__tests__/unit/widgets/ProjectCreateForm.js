import ProjectCreateForm from '../../../src/widgets/ProjectCreateForm.js';
import disabled from '../../../src/directives/disabled';
import directives from '../../../src/directives/event-listener';
import renderer from 'sham-ui-test-helpers';

it( 'renders correctly', () => {
    const meta = renderer( ProjectCreateForm, {
        directives: {
            disabled,
            ...directives
        }
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
