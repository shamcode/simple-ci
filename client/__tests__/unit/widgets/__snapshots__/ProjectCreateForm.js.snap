// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "ID": "widget",
  "constructorOptions": Object {},
  "html": "<div class=\\"project-create-form\\"><form><div class=\\"group\\"><input name=\\"name\\" type=\\"text\\" required=\\"\\"><span class=\\"highlight\\"></span><span class=\\"bar\\"></span><label>Name</label></div><div class=\\"group\\"><input name=\\"cwd\\" type=\\"text\\" required=\\"\\"><span class=\\"highlight\\"></span><span class=\\"bar\\"></span><label>Current Work Directory</label></div><div class=\\"group\\"><button type=\\"submit\\"><span>Create</span></button></div></form></div>",
  "nodes": Array [
    Object {
      "ID": undefined,
      "constructorOptions": null,
      "html": null,
      "nodes": null,
      "options": null,
    },
  ],
  "options": Object {
    "actionSequence": Array [
      "render",
      "bindEvents",
    ],
    "renderAsync": false,
    "renderDependence": Array [],
    "types": Array [],
  },
}
`;
